fiftyone/core/collections.py

```diff
@@ -916,26 +916,45 @@ def _get_per_sample_frames_bytes(self):
         results = self._aggregate(pipeline=pipeline, attach_frames=True)
         return {str(r["_id"]): r["size_bytes"] for r in results}
 
+    def _get_first(self, limit=1, field="_id", reverse=False):
+        direction = -1 if reverse else 1
+        pipeline = [{"$sort": {field: direction}}, {"$limit": limit}]
+
+        return self._aggregate(
+            pipeline=pipeline, detach_frames=True, detach_groups=True
+        )
+
     def first(self):
         """Returns the first sample in the collection.
 
         Returns:
             a :class:`fiftyone.core.sample.Sample` or
             :class:`fiftyone.core.sample.SampleView`
         """
+        cursor = self._get_first()
+
         try:
-            return next(iter(self))
+            d = next(cursor)
         except StopIteration:
             raise ValueError("%s is empty" % self.__class__.__name__)
 
+        return self._make_sample(d)
```

Comment:
Do users use these operations like `first` and `last` often enough that we can implement some caching here too?

Reply:
I don't know about other users, but personally I don't think we'd want to cache the results. I usually call first and last after some mutation as a sanity check